app/bot.py

```diff
@@ -0,0 +1,128 @@
+import asyncio
+import datetime as dt
+import os
+import re
+import sys
+from typing import TYPE_CHECKING, Any, cast, final, override
+
+import discord as dc
+from discord.ext import commands
+from loguru import logger
+
+from app.errors import handle_error, interaction_error_handler
+from app.status import BotStatus
+from app.utils import try_dm
+
+if TYPE_CHECKING:
+    from githubkit import GitHub, TokenAuthStrategy
+
+    from app.config import Config, WebhookFeedType
+
+
+EMOJI_REGEX = re.compile(r"<(a?):(\w+):(\d+)>", re.ASCII)
+
+
+@final
+class GhosttyBot(commands.Bot):
+    def __init__(self, config: "Config", gh: "GitHub[TokenAuthStrategy]") -> None:
+        intents = dc.Intents.default()
+        intents.members = True
+        intents.message_content = True
+        super().__init__(
+            command_prefix=commands.when_mentioned_or("!"),
+            intents=intents,
+            allowed_mentions=dc.AllowedMentions(everyone=False, roles=False),
+        )
+
+        self.tree.on_error = interaction_error_handler
+        self.config = config
+        self.gh = gh
+        self.ghostty_status = BotStatus()
+        self.background_tasks = set[asyncio.Task[None]]()
+
+    @override
+    async def on_error(self, event_method: str, /, *args: Any, **kwargs: Any) -> None:
+        handle_error(cast("BaseException", sys.exception()))
+
+    @override
+    async def setup_hook(self) -> None:
+        await asyncio.gather(
+            *(
+                self.load_extension(ext)
+                for ext in (
+                    f"app.components.{filename.rstrip('.py')}"
+                    for filename in os.listdir("./app/components")  # noqa: PTH208
+                    if not filename.startswith("_")
+                )
+            )
+        )
```

Comment:
1. This is extremely nested and would benefit from using a variable to store the extensions.
2. Use pathlib instead of suppressing the lint :).
3. `./app` feels icky to me (though @trag1c may think otherwise).

```suggestion
        extensions = (
            f"app.components.{file.stem}"
            for file in (Path(__file__).parent / "components").iterdir()
            if not file.name.startswith("_")
        )
        await asyncio.gather(*map(self.load_extension, extensions))
```